### PR TITLE
pocket delivery layer

### LINKED ISSUE #90: pocket delivery layer 추가

### PR DESCRIPTION
**Pull request by BLoHny**
# summary
- add pocket delivery layer
# changes
- add pocket router
- add pocket router test

### CODE REVIEW

File: internal/app/pocket/delivery/pocket_router.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	Offset int `form:"offset" binding:"number"`
+	Limit  int `form:"limit" binding:"number"`
+}
+
+type PocketRequest struct {
+	ReceiverID uint64 `json:"receiverID" binding:"required"`
+	Coins      int    `json:"coins" binding:"number"`
+	Message    string `json:"message" binding:"required"`
+	IsPublic   bool   `json:"isPublic" binding:"required"`
+}
```
Comment (onee-only) [inline internal/app/pocket/delivery/pocket_router.go]: binding required 명시 해 주면 좋겠습니다~

File: internal/app/pocket/delivery/pocket_router.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+		ReceiverID: pocketRequest.ReceiverID,
+		Coins:      pocketRequest.Coins,
+		Message:    pocketRequest.Message,
+		IsPublic:   pocketRequest.IsPublic,
+	})
+	if err != nil {
+		c.Error(err)
+		return
+	}
+}
```
Comment (onee-only) [inline internal/app/pocket/delivery/pocket_router.go]: 이쪽은 status가 반환 되지 않는 것 같습니다!

File: internal/app/pocket/delivery/pocket_router.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+type PocketRequest struct {
+	ReceiverID uint64 `json:"receiverID" binding:"required"`
+	Coins      int    `json:"coins" binding:"number"`
+	Message    string `json:"message" binding:"required"`
+	IsPublic   bool   `json:"isPublic" binding:"required"`
+}
+
+type Visible struct {
+	Visible bool `json:"visible" binding:"required"`
+}
```
Comment (onee-only) [inline internal/app/pocket/delivery/pocket_router.go]:
```suggestion
type VisibilityRequest struct {
	Visible bool `json:"visible" binding:"required"`
}
```
visibilityRequest도 괜찮을 것 같습니다

File: internal/app/pocket/delivery/pocket_router.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+		c.Error(status.NewError(http.StatusBadRequest, "not valid request"))
+		return
+	}
+
+	err = p.pocketUseCase.SetVisibility(c.Request.Context(),
+		&input.VisibilityInput{PocketID: pocketID, Visible: visible.Visible})
+	if err != nil {
+		c.Error(err)
+		return
+	}
```
Comment (onee-only) [inline internal/app/pocket/delivery/pocket_router.go:160]: 여기도 status 보내는 부분 빠졌습니다

File: internal/app/pocket/delivery/pocket_router.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	if err != nil {
+		c.Error(status.NewError(http.StatusBadRequest, "not valid"))
+		return
+	}
+
+	err = p.pocketUseCase.RevealSender(c.Request.Context(), &input.PocketIDInput{PocketID: pocketID})
+	if err != nil {
+		c.Error(err)
+		return
+	}
```
Comment (onee-only) [inline internal/app/pocket/delivery/pocket_router.go:178]: 여기도 status 빠졌네요

File: internal/app/pocket/delivery/pocket_router.go (hunk 1 above)
Comment (onee-only) [reply internal/app/pocket/delivery/pocket_router.go]: [204cd60](https://github.com/lucky-pocket/luckyPocket-back/pull/98/commits/204cd6078a165d4448b93a0ad0a2e6e12145fc6d)

File: internal/app/pocket/delivery/pocket_router.go (hunk 5 above)
Comment (BLoHny) [reply internal/app/pocket/delivery/pocket_router.go:178]: https://github.com/lucky-pocket/luckyPocket-back/pull/98/commits/699b8e3ec52168985d9c79693cf523e2f8705150

File: internal/app/pocket/delivery/pocket_router.go (hunk 4 above)
Comment (BLoHny) [reply internal/app/pocket/delivery/pocket_router.go:160]: https://github.com/lucky-pocket/luckyPocket-back/pull/98/commits/699b8e3ec52168985d9c79693cf523e2f8705150

File: internal/app/pocket/delivery/pocket_router.go (hunk 2 above)
Comment (BLoHny) [reply internal/app/pocket/delivery/pocket_router.go]: https://github.com/lucky-pocket/luckyPocket-back/pull/98/commits/699b8e3ec52168985d9c79693cf523e2f8705150

Comment (onee-only) [reply internal/app/pocket/delivery/pocket_router.go]: status created는 어떤가요?

Comment (BLoHny) [reply internal/app/pocket/delivery/pocket_router.go]: https://github.com/lucky-pocket/luckyPocket-back/pull/98/commits/ae796b82ebe7bf087dfa508ad54695a0c0681f1d

**Review by onee-only — APPROVED** [review]

**Pull request merged**